### PR TITLE
feat: add gem fiat ramp integration

### PR DESCRIPTION
**Pull request by sivaganeshan**
## Description

Please describe your changes

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. etc...

### CODE REVIEW

File: src/pages/BuySell/BuySell.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      return coinifyResponse
+    } catch (ex: any) {
+      throw 'Unknown error detedcted while fetching coinify supported currencies'
+    }
+  }
+
+  const getWyreSupportedCurrencies = async () => {
+    try {
+      let wyreResponse = await (
+        await fetch('https://api.gem.co/institutions/wyre/supported_currencies')
```
Comment (0xdef1cafe) [inline src/pages/BuySell/BuySell.tsx:33]: env var please

File: src/pages/BuySell/BuySell.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      ).json()
+
+      return wyreResponse
+    } catch (ex: any) {
+      throw 'Unknown error detedcted while fetching wyre supported currencies'
```
Comment (0xdef1cafe) [inline src/pages/BuySell/BuySell.tsx:38]: typo in error message, plus same comments as above

File: src/pages/BuySell/BuySell.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  //Filter for the assets you can buy
+  const buyFilter = arg =>
+    arg.transaction_direction === 'bank_blockchain' ||
+    arg.transaction_direction === 'card_blockchain'
+
+  // Filter for the assets you can sell
+  const sellFilter = arg => arg.transaction_direction === 'blockchain_bank'
+
+  //Filter and merge function
+  const filterAndMerge = (arr1, arr2, key, filter) => {
```
Comment (0xdef1cafe) [inline src/pages/BuySell/BuySell.tsx:50]: please wrap this in `useMemo` to avoid rerenders

File: src/pages/BuySell/BuySell.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                    minWidth={'117%'}
+                    >
+                        <Avatar
+                            src={gemlogo}
+                            bg={useColorModeValue('gray.200', 'gray.700')}
+                         />
+                        <Box textAlign='left'>
+                            <Text lineHeight={1}>GEM</Text>
+                            <Text fontWeight='normal' fontSize='sm' >
+                            Buy or sell Crypto with Gem
```
Comment (0xdef1cafe) [inline src/pages/BuySell/BuySell.tsx:138]: translations

File: src/pages/BuySell/BuySell.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+                            </Text>
+                        </Box>
+                        <Flex flexDirection={'row'}  marginLeft={'15%'} >
+                        <Text fontSize={'sm'} color={'green.400'}  borderRadius={'10%'} bg={useColorModeValue('green.200', 'green.800')} padding={'2px'} >
+                                BUY
+                            </Text>
+                            <Text marginLeft={'15%'} fontSize={'sm'} color={'red.400'} bg={useColorModeValue('red.200', 'red.800')} borderRadius={'10%'} padding={'2px'}>
+                                SELL
```
Comment (0xdef1cafe) [inline src/pages/BuySell/BuySell.tsx:146]: translations

File: src/pages/BuySell/BuySell.tsx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+                    minWidth={'115%'}
+                    >
+                        <Avatar
+                            src={onjunologo}
+                            bg={useColorModeValue('gray.200', 'gray.700')}
+                         />
+                        <Box textAlign='left'>
+                            <Text lineHeight={1}>OnJuno</Text>
+                            <Text fontWeight='normal' fontSize='sm'>
+                            Coming soon ...
```
Comment (0xdef1cafe) [inline src/pages/BuySell/BuySell.tsx:173]: translations - @SS-FOX-McCloud do we want this whole coming soon for onjuno?

File: src/pages/BuySell/GemBuySell.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+                    <Flex
+                      justifyContent={'flex-start'}
+                      flexDirection={'column'}
+                      alignItems={'flex-start'}
+                    >
+                      <Text lineHeight={1}>Assest to Buy</Text>
+                      <Box marginTop={'5%'}>
+                        <Menu>
+                          <MenuButton as={Button} rightIcon={<ChevronDownIcon />}>
+                            Select an assest to buy
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:49]: translations

File: src/pages/BuySell/GemBuySell.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+                                      variant='ghost'
+                                      isActive={true}
+                                      justifyContent='flex-start'
+                                      _focus={{
+                                        shadow: 'outline-inset'
+                                      }}
+                                      padding={'25px'}
+                                    >
+                                      <Avatar
+                                        src={`https://gem-widgets-assets.s3-us-west-2.amazonaws.com/currencies/crypto/${buyCoin.ticker.toLowerCase()}.svg`}
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:66]: env vars

File: src/pages/BuySell/GemBuySell.tsx [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                      </Box>
+                    </Flex>
+                    <Flex
+                      justifyContent={'center'}
+                      flexDirection={'column'}
+                      alignItems={'center'}
+                      marginTop={'15%'}
+                    >
+                      <Box>
+                        <Button>Continue</Button>
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:91]: translations

File: src/pages/BuySell/GemBuySell.tsx [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+                      </Box>
+                      <Box marginTop={'5%'}>
+                        <Button>Cancel</Button>
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:94]: translations

File: src/pages/BuySell/GemBuySell.tsx [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+                      </Box>
+                    </Flex>
+                    <Flex
+                      justifyContent={'center'}
+                      flexDirection={'column'}
+                      alignItems={'center'}
+                      marginTop={'15%'}
+                    >
+                      <Box>
+                        <Button>Continue</Button>
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:150]: translations

File: src/pages/BuySell/GemBuySell.tsx [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+                      </Box>
+                      <Box marginTop={'5%'}>
+                        <Button>Cancel</Button>
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:153]: translations

File: src/pages/BuySell/GemBuySell.tsx [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+                            })}
+                          </MenuList>
+                        </Menu>
+                      </Box>
+                    </Flex>
+                    <Flex
+                      justifyContent={'center'}
+                      flexDirection={'column'}
+                      alignItems={'center'}
+                      marginTop={'15%'}
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:147]: i think we have enums for these margin sizes defined so we're consistent with our styles everywhere

File: src/pages/BuySell/GemBuySell.tsx [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+  Tabs,
+  Text,
+  useColorModeValue
+} from '@chakra-ui/react'
+import { useLocation } from 'react-router-dom'
+import { Card } from 'components/Card/Card'
+import { Page } from 'components/Layout/Page'
+import { Select } from '@chakra-ui/react'
+
+export default function GemBuySell() {
```
Comment (0xdef1cafe) [inline src/pages/BuySell/GemBuySell.tsx:24]: please avoid default exports

File: src/pages/BuySell/BuySell.tsx [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+
+  // Filter for the assets you can sell
+  const sellFilter = arg => arg.transaction_direction === 'blockchain_bank'
+
+  //Filter and merge function
+  const filterAndMerge = (arr1, arr2, key, filter) => {
+    const list1 = arr1.filter(filter).map(list => list[key].currencies)
+    const list2 = arr2.filter(filter).map(list => list[key].currencies)
+    const results = list1.concat(list2).flat()
+    let filteredResults = []
```
Comment (0xdef1cafe) [inline src/pages/BuySell/BuySell.tsx:54]: please use a reduce here rather than mutating

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
hey @sivaganeshan thanks for the PR, this is a good first pass. i've left a bunch of comments to be addressed.

can you please run `yarn lint:fix` to get things formatted nicely? you can also configure your editor to respect the `.eslintrc` file in the root of the repo so it automatically formats on save.

**Comment by sivaganeshan** [comment]
will work on the changes and update the PR

**Comment by 0xdef1cafe** [comment]
additionally, @sivaganeshan can you please fill out the PR template, including some screenshots?

**Comment by 0xdef1cafe** [comment]
hey @sivaganeshan - quite a few unresolved comments. are you able to resolve the rest please?

**Comment by sivaganeshan** [comment]
> 0xdef1cafe

working on the same

**Comment by 0xdef1cafe** [comment]
> > 0xdef1cafe
> 
> working on the same

thanks @sivaganeshan - do you have an estimated completion date for this PR?

**Comment by sivaganeshan** [comment]
> > > 0xdef1cafe
> > 
> > 
> > working on the same
> 
> thanks @sivaganeshan - do you have an estimated completion date for this PR?

Hi  @0xdef1cafe ,  I will be finishing all the review comments by tomorrow, and the gem parameters formation is still pending from me to work on.  Required gem sandbox parameters to work on the same

**Comment by 0xdef1cafe** [comment]
hey @sivaganeshan - thanks for the contribution but as discussed with josh in discord we are going to assign this to another contributor at this point. please speak with @0xean on or after 2/14 with respect to compensation for this bounty.

**Comment by 0xean** [comment]
ref: #847

**Pull request closed without merging**